app/models/notification.rb

```diff
@@ -0,0 +1,41 @@
+class Notification < ActiveRecord::Base
+  belongs_to :notification_type
+  belongs_to :initiator, :class_name => User, :foreign_key => 'user_id'
+  belongs_to :subject, :polymorphic => true
+  has_many :notification_recipients, :dependent => :delete_all
+  has_many :recipients, :class_name => User, :through => :notification_recipients, :source => :user
+
+  validates :notification_type, :presence => true
+  validates :initiator, :presence => true
```

Comment:
it simply marks who initated the even that lead to the notification, think user X deleted a whole bunch of stuff and now you need to be aware of that.
or another usage case, your org admin (user x) created a new hostgroup that you can now use.